### PR TITLE
List resource groups for AKS

### LINKED ISSUE #10898: List AKS Resource Groups
### Description of the feature you would like to add / User story

As a user 
I would like to have a list of Azure Resource Groups
in order to use that while cluster creation

### Solution details

- Requirement 1
- Requirement 2
- Requirement 3

### PR DESCRIPTION
**Pull request by imharshita**
Signed-off-by: Harshita sharma <harshita.sharma6174@gmail.com>

**What this PR does / why we need it**: List resource groups in an Azure subscription.

**Which issue(s) this PR fixes**:

Fixes #10898

**What type of PR is this?**

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Adds GET endpoint `/api/v2/providers/aks/resourcegroups` to list AKS resource groups
```

**Documentation**:

```documentation
TBD
```

### CODE REVIEW

**Comment by ahmedwaleedmalik** [comment]
@imharshita Please update the PR title to be more specific; list resource groups for AKS. Also, update the description and add a release note that a new endpoint is introduced for listing resource groups for an AKS cluster or something similar.

File: cmd/kubermatic-api/swagger.json [hunk 1]
```diff
@@ -21818,7 +21844,7 @@
     },
     "AKSLocation": {
       "type": "object",
-      "title": "AKSLocation is the object representing Azure Location.",
+      "title": "AKSLocation represents an object representing Azure Location.",
```
Comment (ahmedwaleedmalik) [inline cmd/kubermatic-api/swagger.json:21847]: `AKSLocation represents an object representing Azure Location.` Please re-word this

File: pkg/api/v2/types.go [hunk 2]
```diff
@@ -945,7 +945,7 @@ type AKSVMSizeList []AKSVMSize
 // swagger:model AKSLocationList
 type AKSLocationList []AKSLocation
 
-// AKSLocation is the object representing Azure Location.
+// AKSLocation represents an object representing Azure Location.
```
Comment (ahmedwaleedmalik) [inline pkg/api/v2/types.go]: > AKSLocation represents an object representing Azure Location.

Please reword this.

**Comment by imharshita** [comment]
@ahmedwaleedmalik ptal

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Comment by imharshita** [comment]
cc @KhizerRehan

**Pull request merged**